server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentCompactor.java

```diff
@@ -73,32 +76,24 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
         Map<String, DataSourceCompactionConfig> compactionConfigs = compactionConfigList
             .stream()
             .collect(Collectors.toMap(DataSourceCompactionConfig::getDataSource, Function.identity()));
-        final int numRunningCompactTasks = indexingServiceClient
-            .getRunningTasks()
-            .stream()
-            .filter(status -> {
-              final String taskType = status.getType();
-              // taskType can be null if middleManagers are running with an older version. Here, we consevatively regard
-              // the tasks of the unknown taskType as the compactionTask. This is because it's important to not run
-              // compactionTasks more than the configured limit at any time which might impact to the ingestion
-              // performance.
-              return taskType == null || taskType.equals(COMPACT_TASK_TYPE);
-            })
-            .collect(Collectors.toList())
-            .size();
+        final int numNonCompleteCompactionTasks = findNumNonCompleteCompactTasks(
+            indexingServiceClient.getRunningTasks(),
+            indexingServiceClient.getPendingTasks(),
+            indexingServiceClient.getWaitingTasks()
+        );
         final CompactionSegmentIterator iterator = policy.reset(compactionConfigs, dataSources);
 
         final int compactionTaskCapacity = (int) Math.min(
             indexingServiceClient.getTotalWorkerCapacity() * dynamicConfig.getCompactionTaskSlotRatio(),
             dynamicConfig.getMaxCompactionTaskSlots()
         );
-        final int numAvailableCompactionTaskSlots = numRunningCompactTasks > 0 ?
-                                                    compactionTaskCapacity - numRunningCompactTasks :
+        final int numAvailableCompactionTaskSlots = numNonCompleteCompactionTasks > 0 ?
+                                                    compactionTaskCapacity - numNonCompleteCompactionTasks :
                                                     // compactionTaskCapacity might be 0 if totalWorkerCapacity is low.
                                                     // This guarantees that at least one slot is available if
                                                     // compaction is enabled and numRunningCompactTasks is 0.
                                                     Math.max(1, compactionTaskCapacity);
-        LOG.info("Running tasks [%d/%d]", numRunningCompactTasks, compactionTaskCapacity);
+        LOG.info("Running tasks [%d/%d]", numNonCompleteCompactionTasks, compactionTaskCapacity);
```

Comment:
can we make this more readable something like ` Total running and waiting tasks [], Maximum compaction queue capacity [] `. Thanks

Reply:
Good point. Fixed.